### ISSUE TITLE
Error on Android When getting the current location with enableHighAccuracy : true

### ISSUE THREAD

**Issue by viroja123** [michalchudziak/react-native-geolocation#343]
![Screenshot from 2024-11-29 18-39-32](https://github.com/user-attachments/assets/307f11eb-cbc4-4a32-9954-750e6f96ea78)
![Screenshot from 2024-11-29 18-40-00](https://github.com/user-attachments/assets/14473cd8-1705-45df-8dec-250f082a5bab)
## Versions

- Android:
- iOS:
- react-native-geolocation:
- react-native:
- react:

**Comment by anuragchachan** [comment]
yes i am also facing same issue , when enableHighAccuracy : true then it takes longer time and at last results in error.

**Comment by hashhirr** [comment]
same issue i had to set enablehHighAccuracy to FALSE to make it work but the location is not accurate

**Comment by anuragchachan** [comment]
i tried downgrading to version 3.3.0 , its working in enableHighAccuracy : true but accuracy is not correct.

**Comment by hashhirr** [comment]
> i tried downgrading to version 3.3.0 , its working in enableHighAccuracy : true but accuracy is not correct.

there is another library react-native-background-geolocation but it's paid

**Comment by SebastianNarvaez11** [comment]
Same issue here

**Comment by muffix200** [comment]
I also had these issues. I just moved [this](https://www.npmjs.com/package/react-native-get-location) library, and it's working fine

**Comment by AryanBansal21** [comment]
can anyone please tell why by doing enableHighAccuracy : false it works fine

**Comment by dgreasi** [comment]
I have the same issue for Android:
```
"@react-native-community/geolocation": "^3.4.0",
"react-native": "0.73.11",
```

As a workaround I am requesting initially the precise location and if that times-out I request it with `enableHighAccuracy: false`.

Example:
```
export async function getCurrentPosition(): Promise<CurrentPosition | null> {
  const highAccuracy = await getPosition();
  console.log('HIGH ACCURACY LOCATION RESOLVED');
  if (highAccuracy) return highAccuracy;

  console.log('**UNABLE** TO GET HIGH ACCURACY LOCATION');

  const coarceAccuracy = await getPosition(false);
  return coarceAccuracy;
}


function getPosition(enableHighAccuracy = true, timeout = 15000, maximumAge = 0): Promise<CurrentPosition | null> {
  return new Promise(function (resolve) {
    Geolocation.getCurrentPosition(
      (position: GeolocationResponse) => {
        resolve({
          latitude: position?.coords?.latitude,
          longitude: position?.coords?.longitude,
          timestamp: position?.timestamp,
        });
      },
      (error) => {
        console.log('getPosition - error: ', { error, enableHighAccuracy });
        resolve(null);
      },
      {
        enableHighAccuracy,
        timeout,
        maximumAge,
      },
    );
  });
}
```

**Comment by rajush1** [comment]
react-native-geolocation-service
React native geolocation service for iOS and android.

Why ?
This library is created in an attempt to fix the location timeout issue on android with the react-native's current implementation of Geolocation API. This library tries to solve the issue by using Google Play Service's new FusedLocationProviderClient API, which Google strongly recommends over android's default framework location API. It automatically decides which provider to use based on your request configuration and also prompts you to change the location mode if it doesn't satisfy your current request configuration.
